src/figuresClasses.ts

```diff
@@ -1,19 +1,78 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+function isLessThanZero(arg: number): boolean {
+  return arg <= 0;
 }
 
-export class Triangle {
-
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if ([a, b, c].some(isLessThanZero)) {
+      throw new Error('Invalid side length');
+    }
+
+    if ((a + b) <= c || (a + c) <= b || (b + c) <= a) {
+      throw new Error('It\'s not a triangle');
+    }
+  }
 
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) / 2;
+
+    return +Math
+      .sqrt(p * (p - this.a) * (p - this.b) * (p - this.c))
+      .toFixed(2);
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if ([this.radius].some(isLessThanZero)) {
```

Comment:
You don't need to write `this` inside of the constructor. Rewrite all over the code.
```suggestion
    if ([radius].some(isLessThanZero)) {
```